Observer/Customer/SaveAfterData.php

```diff
@@ -45,89 +48,76 @@ public function __construct(
-            $platformCustomer = null;
+            $this->savePlatformCustomer($customer);
         } catch (\Exception $e) {
             $this->logger->critical($e);
-            $platformCustomer = null;
         }
+    }
 
-        return $platformCustomer;
+    /**
+     * @param \Magento\Customer\Api\Data\CustomerInterface $customer
+     * @param \Magento\Customer\Api\Data\CustomerInterface $origCustomer
+     * @return bool
+     */
+    private function isCustomerChanged(CustomerInterface $customer, CustomerInterface $origCustomer): bool
+    {
+        return $origCustomer->getEmail() !== $customer->getEmail()
+            || $origCustomer->getFirstname() !== $customer->getFirstname()
+            || $origCustomer->getLastname() !== $customer->getLastname()
+            || $origCustomer->getGroupId() !== $customer->getGroupId();
     }
 
     /**
-     * @param CustomerInterface $customer
-     * @param PlatformCustomerInterface $platformCustomer
+     * @param \Magento\Customer\Api\Data\CustomerInterface $customer
+     * @return void
      */
-    protected function updatePlatformCustomer(
-        CustomerInterface $customer,
-        PlatformCustomerInterface $platformCustomer
-    ): void {
-        try {
-            $this->importCustomerData($platformCustomer, $customer);
-            $this->platformCustomerService->saveCustomer($platformCustomer, $customer->getWebsiteId());
-        } catch (\Exception $e) {
-            $this->logger->critical($e);
+    private function savePlatformCustomer($customer): void
+    {
+        $websiteId = $customer->getWebsiteId();
+
+        $platformCustomer = $this->platformCustomerManager->getCustomerByMagentoCustomerId(
+            (int)$customer->getId(),
+            $this->isCreateNewCustomerIfNotExist($websiteId),
+            $websiteId
+        );
+
+        if ($platformCustomer) {
+            $platformCustomer->setFirstName($customer->getFirstname());
+            $platformCustomer->setLastName($customer->getLastname());
+            $platformCustomer->setEmail($customer->getEmail());
+            $platformCustomer->setMagentoCustomerGroupId($customer->getGroupId());
+
+            $this->platformCustomerService->saveCustomer($platformCustomer, (int)$websiteId);
         }
     }
 
     /**
-     * @param PlatformCustomerInterface $platformCustomer
-     * @param CustomerInterface $customer
+     * @param int|string|null $websiteId
+     * @return bool
      */
-    protected function importCustomerData(
-        PlatformCustomerInterface $platformCustomer,
-        CustomerInterface $customer
-    ): void {
-        $platformCustomer->setFirstName($customer->getFirstname());
-        $platformCustomer->setLastName($customer->getLastname());
-        $platformCustomer->setEmail($customer->getEmail());
-        $platformCustomer->setMagentoCustomerGroupId($customer->getGroupId());
+    private function isCreateNewCustomerIfNotExist($websiteId): bool
```

Comment:
Please, define `$websiteId` argument type

Reply:
Could not resolve as multiple types are not supported in  php7 type hints